### DISCUSSION TITLE
Help to Run the Project

### DISCUSSION THREAD

**Discussion by ParthibanSubramaniyan** [NoahHsu/event-sourcing-order-poc#76] [General]
I am learning event sourcing and found this project. Thanks for this.  I see https://github.com/NoahHsu/event-sourcing-order-poc/blob/master/backstage/docs/run.md 

# run all command, query side, and dependencies 
docker compose -f Docker/observe-docker-compose.yaml -f Docker/boot-run-order-apps-docker-compose.yml -f Docker/boot-run-payment-apps-docker-compose.yml -f Docker/boot-run-shipment-apps-docker-compose.yml -f Docker/kafka-docker-compose.yml --env-file Docker/config/.env.docker -p event-sourcing up

I don't see these file under Docker directory.
 Docker/observe-docker-compose.yaml 
Docker/boot-run-order-apps-docker-compose.yml 
Docker/boot-run-payment-apps-docker-compose.yml 
Docker/boot-run-shipment-apps-docker-compose.yml 
Docker/kafka-docker-compose.yml 

Are there any files are not checked in? Can you please help understand this?

**Comment by NoahHsu** [comment]
Sorry for misleading you, I found it so troublesome to run like this way, 
so I open this [PR](https://github.com/NoahHsu/event-sourcing-order-poc/pull/74) to refactor how to run the App.
and delete the related files (the files you mention above)

But I forget to update the instruction in [backstage/docs/run.md](https://github.com/NoahHsu/event-sourcing-order-poc/blob/master/backstage/docs/run.md)

so, could you try to run it by your IDE run Application or sth like `./gradlew :order:order-command-side:bootRun`
I quickly update the way in a new [PR](https://github.com/NoahHsu/event-sourcing-order-poc/pull/77)
please refer to it, and see whether it works for you

thank you for leaving this message~

**Comment by ParthibanSubramaniyan** [comment]
Thank you very much for responding. I was able to run command side & query side on all order, payment, shipping.
I am facing issue starting event-handler of order

Getting the below error
{"@timestamp":"2024-12-03T14:16:12.028-08:00","@version":"1","message":"Application run failed","logger":"org.springframework.boot.SpringApplication","thread":"main","level":"ERROR","levelValue":40000,"stackTrace":"j.l.IllegalArgumentException: Could not resolve placeholder 'cluster.order.command.base-url' in value \"${cluster.order.command.base-url}\"

having problem with payment & shipment event handler bootRun as well. 

Is it event handler across order, payment, shipment working for you? Can you please advise?

**Reply by NoahHsu** [reply]
yeah, my recent change didn't modify the handler application.

so it won't work for now. I will find a time to fix it. 
I open an [issue](https://github.com/NoahHsu/event-sourcing-order-poc/issues/78) as a record.
Thank you for reminding this part, I almost forgot it.

**Reply by ParthibanSubramaniyan** [reply]
Thank you
